### PR TITLE
Fix for issue 42 ( Add maps.me mwm export )

### PR DESCRIPTION
**Pull request by unithiyashree**
I have used the binaries provided by Zverik and added mwm. format to export tool as my first contribution task to OPW intern .Please review it and revert back if anything goes wrong. "

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -163,7 +178,9 @@ Look at <code>core/settings/project.py</code> and make sure you update or overri
 
 **GARMIN_CONFIG** = 'absolute path to utils/conf/garmin_config.xml'
 
-**OVERPASS_API_URL** = 'url of your local overpass api endpoint (see Overpass API below)'
+**OVERPASS_API_URL** = 'http://overpass-api.de/api/interpreter'
```
Comment (pnorman) [inline README.md]: Wouldn't this cause the export tool to pull from the public overpass API?

**Comment by pnorman** [comment]
You've included binaries for MWM. This is different from mkgmap and splitter, where the [readme instructs to download them separately](https://github.com/hotosm/osm-export-tool2#garmin). It's probably best to handle mwm binaries like garmin binaries.

------

The readme states that "[The HOT Exports service uses a local instance of Overpass v07.52 for data extraction](https://github.com/hotosm/osm-export-tool2#overpass-api)" but some of your config changes would have it using the public instance, not the local one. If this was intentional, it should go in its own PR for independent review as it's not related to MWM.

File: README.md [hunk 2]
```diff
@@ -108,6 +109,20 @@ The HOT Export pipeline depends on a number of third-party tools.
...
 
+Install Qt 5.5:
+
+<code>$ sudo add-apt-repository ppa:beineri/opt-qt55-trusty </code>
+
+<code>$ sudo apt-get update </code>
+
+<code>$ sudo apt-get install qt55base </code>
+
+<code>$ source /opt/qt55/bin/qt55-env.sh </code>
```
Comment (dodobas) [inline README.md]: It's not clear why is Qt5.5 needed, and why are you adding something to the environment with `source` as it will be lost when you close theterminal session

File: README.md (hunk 1 above)
Comment (dodobas) [inline README.md]: Please remove this, you can probably use public overpass api server for 'testing', and it could be added to the documentation, but not like this

File: README.md [hunk 3]
```diff
@@ -197,14 +214,14 @@ they are pushed to a Celery Worker for processing. At least two celery workers n
...
-<code>export DJANGO_SETTINGS_MODULE=core.settings.your_settings_module</code>
+<code>export DJANGO_SETTINGS_MODULE=core.settings.your_settings_module</code>  [ your_settings_module is dev ]
 
 <code>$ celery -A core worker --loglevel debug --logfile=celery.log</code>.
 
 This will start a celery worker which will process export tasks. An additional celery worker needs to be started to handle purging of expired unpublished
 export jobs. From another hotosm virtualenv terminal session in the project top-level directory, run:
 
-<code>export DJANGO_SETTINGS_MODULE=core.settings.your_settings_module</code>
+<code>export DJANGO_SETTINGS_MODULE=core.settings.your_settings_module</code>  [ your_settings_module is dev ]
```
Comment (dodobas) [inline README.md]: Documentation lacks some info on how the project is structured, and how a developer should prepare *a working development environment*, It would be better to remove these additions

File: core/settings/project.py [hunk 4]
```diff
@@ -36,11 +37,14 @@
...
 
+# the root url for maps me binary files
+EXPORT_MWM_ROOT = ''
+
 # location of the garmin config file
 GARMIN_CONFIG = '/home/ubuntu/www/hotosm/utils/conf/garmin_config.xml'
 
 # url to overpass api endpoint
-OVERPASS_API_URL = 'http://localhost/interpreter'
+OVERPASS_API_URL = 'http://overpass-api.de/api/interpreter'
```
Comment (dodobas) [inline core/settings/project.py]: please revert this change, public overpass api can be used for testing

File: jobs/migrations/insert_export_formats.py [hunk 5]
```diff
@@ -11,6 +11,8 @@ class Migration(migrations.Migration):
 
     def insert_export_formats(apps, schema_editor):
         ExportFormat = apps.get_model('jobs', 'ExportFormat')
+        ExportFormat.objects.create(name='MWM Format', description='Maps.me File format',
+                                    slug='MWM')
```
Comment (dodobas) [inline jobs/migrations/insert_export_formats.py]: this is a problem, because you are changing a migration file, so, if an osm-export-tool2 installation like (http://export.hotosm.org) has already applied migrations, this change cannot be applied, which is a problem

you need to create a new custom migration and add MWM setup to that migration

File: utils/mwm.py [hunk 6]
```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8 -*-
+from __future__ import with_statement
```
Comment (dodobas) [inline utils/mwm.py]: this is only needed if you use Python 2.5 ... please remove

File: mapsme_generator/data/car.lua [hunk 7]
```diff
@@ -0,0 +1,437 @@
+-- Begin of globals
```
Comment (dodobas) [inline mapsme_generator/data/car.lua]: please remove all binary files which are not specifically changed to work with `osm-export-tool2`, and which are part of basic maps.me installation

File: README.md [hunk 8]
```diff
@@ -62,7 +62,8 @@ Install PostgreSQL / PostGIS and its dependencies,
 <pre>
 $ sudo su - postgres
 $ createdb 'hot_exports_dev'
-$ create role hot with password '<-password->'
+$ psql
```
Comment (dodobas) [inline README.md]: `psql -c "create role hot with password '<-password->'" database_name`

File: README.md [hunk 9]
```diff
@@ -163,7 +178,9 @@ Look at <code>core/settings/project.py</code> and make sure you update or overri
 
 **GARMIN_CONFIG** = 'absolute path to utils/conf/garmin_config.xml'
 
-**OVERPASS_API_URL** = 'url of your local overpass api endpoint (see Overpass API below)'
+**OVERPASS_API_URL** = 'http://overpass-api.de/api/interpreter'
+
+**EXPORT_MWM_ROOT** = 'path to  mapsme_generator directory'
```
Comment (dodobas) [inline README.md]: so, how do you actually install this utility, documentation is missing

File: README.md (hunk 2 above)
Comment (unithiyashree) [inline README.md]: https://github.com/hotosm/osm-export-tool2/issues/42 . Referring to the comments made by @Zverik . For the generator tool to work we need it installed. Referred form :- https://github.com/mapsme/omim/blob/master/docs/INSTALL.md

File: README.md (hunk 9 above)
Comment (unithiyashree) [inline README.md]: Where should I document it ?

File: README.md (hunk 2 above)
Comment (Zverik) [inline README.md]: Maybe this line can be moved to `generate_mwm.sh`?
